src/components/Header.jsx

```diff
@@ -0,0 +1,47 @@
+import { Link, NavLink } from "react-router-dom"
+import { useSelector } from 'react-redux'
+
+import '../styles/header.scss'
+
+const Header = ({ searchMovies }) => {
+  
+  const { starredMovies } = useSelector((state) => state.starred)
+
+  return (
+    <header>
+      <Link to="/" data-testid="home" onClick={() => searchMovies('')}>
+        <i className="bi bi-film" />
+      </Link>
+
+      <nav>
+        <NavLink to="/starred" data-testid="nav-starred" className="nav-starred">
+          {starredMovies.length > 0 ? (
+            <>
+            <i className="bi bi-star-fill bi-star-fill-white" />
+            <sup className="star-number">{starredMovies.length}</sup>
+            </>
+          ) : (
+            <i className="bi bi-star" />
+          )}
+        </NavLink>
+        <NavLink to="/watch-later" className="nav-fav">
+          watch later
+        </NavLink>
+      </nav>
+
+      <div className="input-group rounded">
+        <Link to="/" onClick={(e) => searchMovies('')} className="search-link" >
```

Comment:
Having the <input> nested within the <Link> could unintentionally trigger redirection to the home route every time a user interacts with the input field. Removing the `<Link>` wrapper around the `<input>` element should prevent unwanted redirection while retaining the functionality of the search feature.